.github/workflows/build_and_test_feature.yml

```diff
@@ -3,279 +3,280 @@ name: Build & Test Feature
+          yarn --cwd packages/react-component-library lint:ci
+
+      - name: Persist artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: lintreact
+          path: packages/react-component-library/test-results/eslint/react-component-results.xml
+
+  Test_react-component-library:
+    runs-on: ubuntu-latest
+    needs: [Build_icon_library, Lint_react-component-library]
+    steps:
+      - name: Git clone repository
+        uses: actions/checkout@v2
+
+      - name: Cache Node modules
+        uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ secrets.CACHE_VERSION }}-${{ hashFiles('**/yarn.lock') }}
+
+      - name: Attach workspace
+        uses: actions/download-artifact@v2
+        with:
+          name: dist
+
+      - name: Jest
+        env:
+          JEST_JUNIT_OUTPUT_DIR: test-results/jest
+          JEST_JUNIT_OUTPUT_NAME: react-component-results.xml
+        run: |
+          tar -xzf dist.tar.gz && mv distil packages/icon-library/dist && mv distdt packages/design-tokens/dist
+          yarn --cwd packages/react-component-library test --ci --coverage --silent --no-cache --reporters=default --reporters=jest-junit --runInBand --testResultsProcessor=jest-sonar-reporter
+
+      - name: SonarCloud Scan
+        uses: defencedigital/design-system-sonarcloud-action@master
+        if: ${{ github.actor != 'dependabot[bot]' }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+
+  Test-e2e_react-component-library:
+    runs-on: ubuntu-latest
+    needs: [Build_icon_library, Lint_react-component-library]
+    steps:
+      - name: Git clone repository
+        uses: actions/checkout@v2
+
+      - name: Cache Node modules
+        uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ secrets.CACHE_VERSION }}-${{ hashFiles('**/yarn.lock') }}
+
+      - name: Cache Cypress binary
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/Cypress
+          key: cypress-${{ runner.os }}-cypress-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            cypress-${{ runner.os }}-cypress-
+
+      - name: Attach workspace
+        uses: actions/download-artifact@v2
+        with:
+          name: dist
+
+      - name: Unpack icon library & design tokens
+        run: |
+          tar -xzf dist.tar.gz && mv distil packages/icon-library/dist && mv distdt packages/design-tokens/dist
+
+      - name: Confirm Cypress binary is installed
+        run: |
+          result=$(yarn cypress version)
+          if [[  `echo $result` == *"not installed"* ]];
+            then echo "cypress binary is not installed, installing binary...." && yarn cypress install;
+            else echo "cypress binary is installed, no action required";
+          fi
+
+      - name: Run e2e tests - Chrome
```

Comment:
Would be nice to use a matrix for this if we can: https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix